rocketpy/mathutils/function.py

```diff
@@ -2646,6 +2719,152 @@ def compose(self, func, extrapolate=False):
+        interpolation, and extrapolation parameters are compatible with the
+        given source. It converts the source to a numpy array if necessary, sets
+        default values and raises warnings or errors for incompatible or
+        ill-defined parameters.
+
+        Parameters
+        ----------
+        source : list, np.ndarray, or Function
+            The source data or Function object. If a list or ndarray, it should
+            contain numeric data. If a Function, its inputs and outputs are
+            checked against the provided inputs and outputs.
+        inputs : list of str or None
+            The names of the input variables. If None, defaults are generated
+            based on the dimensionality of the source.
+        outputs : str or list of str
+            The name(s) of the output variable(s). If a list is provided, it
+            must have a single element.
+        interpolation : str or None
+            The method of interpolation to be used. For multidimensional sources
+            it defaults to 'shepard' if not provided.
+        extrapolation : str or None
+            The method of extrapolation to be used. For multidimensional sources
+            it defaults to 'natural' if not provided.
+
+        Returns
+        -------
+        tuple
+            A tuple containing the processed inputs, outputs, interpolation, and
+            extrapolation parameters.
+
+        Raises
+        ------
+        ValueError
+            If the dimensionality of the source does not match the combined
+            dimensions of inputs and outputs. If the outputs list has more than
+            one element.
+        TypeError
+            If the source is not a list, np.ndarray, or Function object.
+        Warning
+            If inputs or outputs do not match for a Function source, or if
+            defaults are used for inputs, interpolation,and extrapolation for a
+            multidimensional source.
+
+        Examples
+        --------
+        >>> from rocketpy import Function
+        >>> source = np.array([(1, 1), (2, 4), (3, 9)])
+        >>> inputs = "x"
+        >>> outputs = ["y"]
+        >>> interpolation = 'linear'
+        >>> extrapolation = 'zero'
+        >>> inputs, outputs, interpolation, extrapolation = Function._check_user_input(
+        ...     source, inputs, outputs, interpolation, extrapolation
+        ... )
+        >>> inputs
+        ['x']
+        >>> outputs
+        ['y']
+        >>> interpolation
+        'linear'
+        >>> extrapolation
+        'zero'
+        """
+        # check output type and dimensions
+        if isinstance(outputs, str):
+            outputs = [outputs]
+        if isinstance(inputs, str):
+            inputs = [inputs]
+
+        elif len(outputs) > 1:
+            raise ValueError(
+                "Output must either be a string or have dimension 1, "
+                + f"it currently has dimension ({len(outputs)})."
+            )
```

Comment:
I don't know if this is considered a higher dimensional output, but this code runs fine:

```py
import numpy as np
from rocketpy import Function

def source(x):
    return np.array([x, x+1, x+2])

f = Function(source)

f.plot()
```

![image](https://github.com/RocketPy-Team/RocketPy/assets/87212571/928e58c1-6081-4f24-a42a-0d9323e84811)

I can even differentiate it:

```bash
>>> print(f.differentiate(5))
array([1., 1., 1.])
```

Reply:
![image](https://github.com/RocketPy-Team/RocketPy/assets/63590233/26bb283f-38bf-4ea3-84bd-dbcfeeb29255)

Reply:
Indeed it seems to be a half supported feature, since it cannot be made with array based Functions and there is this odd R1->R1.

The plots work since `matplotlib` supports multiple values on the "y".

I agree moving this PR foward since implementing fully this feature would be out of scope of this PR.

Reply:
Yes, agreed.

Just realized something here... So the "outputs" argument should always be a list of one single string. Right?

And should we flag a TODO do understand (and maybe remove) this ValueError raise in this part of the code?

Reply:
Indeed, if the Function only supported output dimension is one, then the `outputs` argument should only be a list of a single string.

I do think that we can support multidimensional outputs due the fact `numpy.ndarray` bahaves quite well in this cases, with a behavior basically undistinguished from a number (e.g. `array([1,1]) + array([1,1]) => array([2,2])`).

Reply:
Does that kind of Function make sense to you? A mathematical operator that takes 1 value in the domain and associate it to 3 different values (3 different images?).

Reply:
I believe mathematically it is a well defined operation, but cannot see right now an application for it in the code.